line item enhanced updates (#98)
fivetran/dbt_stripe: CHANGELOG.md

# dbt_stripe v0.15.1

## Bug Fixes
- Updated the logic in `stripe__line_item_enhanced` to properly bring in refund data by adjusting the joins on balance transactions, refunds and charges. 
  - Since charges and refunds are both types of balance transactions, included an additional join between refunds and balance transactions to bring in refunds at the same level as charges. 
  - Fixed `fee_amount` logic to sum together charge and refund amounts.

>   - Updated `transaction_type` logic to not only bring in `refund`, but return `charge+refund` if the balance transaction has a charge and a refund associated with it.

Author: Updating. I don't believe a `refund` can exist in Stripe without a `charge` so for now won't add `refund` in ([documentation](https://docs.stripe.com/refunds))

>   - Updated `transaction_type` logic to not only bring in `refund`, but return `charge+refund` if the balance transaction has a charge and a refund associated with it.

Author: Adding `payment intent + refund` as a transaction type based on Jamie's research.

>   - Updated conditional logic for invoice-only records to bring in only non-zero rather than not null `fee_amounts`, as summing charge and refund amounts together brings in only not null `fee_amount` values.

Reviewer: Is the update about the filter or about us coalescing `fee_amount` with 0 now for non-header rows?

>   - Updated conditional logic for invoice-only records to bring in only non-zero rather than not null `fee_amounts`, as summing charge and refund amounts together brings in only not null `fee_amount` values.

Author: Both. Before it was possible for `fee_amount` to be null because of the left join. Because we are summing, summing a null with a not null would make the value null. So I coalesced to zero, meaning it could only be zero or not zero.

# dbt_stripe v0.15.1

## Bug Fixes
- Updated the logic in `stripe__line_item_enhanced` to properly bring in refund data by adjusting the joins on balance transactions, refunds and charges. 
  - Since charges and refunds are both types of balance transactions, included an additional join between refunds and balance transactions to bring in refunds at the same level as charges. 
    - Updated balance transactions join on `connected_account_id` and `source_relation` to look at both charge and refund balance transactions.
  - Fixed `fee_amount` logic to sum together charge and refund amounts.

>     - Updated conditional logic for invoice-only records to ensure `fee_amount` is not null by coalescing to zero for summing non-header rows.

Reviewer: ```suggestion
    - Coalesced `fee_amount` with zero for invoice-only (non-header) rows and updated downstream summing logic accordingly. 
```

Author: Oh yeah that's much clearer. Thanks @fivetran-jamie !
